src/app/App.js

```diff
@@ -61,6 +61,14 @@ class App extends EventEmitter {
 		this.captureScrollPositionFromScrollEvent = true;
 
 		/**
+		 * Holds the value of the current browser path.
+		 * @type {!string}
+		 * @default the current browser path.
+		 * @protected
+		 */
+		this.currentBrowserPath = window.location.pathname;
```

Comment:
I thought you just said to get it from window.location.pathname. We need to store it. On the popstate event the window.location.pathname points to the already modified path. If we need to check if it's a hash change there is no way to unless we know what the previous path was.